src/crab/ebpf_checker.cpp

```
@@ -27,6 +27,65 @@ namespace {
+constexpr int bpf_sock_state_offset = 72;
+constexpr int bpf_sock_rx_queue_mapping_offset = 76;
+constexpr int bpf_sock_u32_field_size = 4;
+
+bool is_power_of_two(const int size) { return size > 0 && (size & (size - 1)) == 0; }
+
+bool is_narrow_sock_field_access(const int offset, const int size, const int field_offset, const int field_size) {
+    return size <= bpf_sock_u32_field_size && is_power_of_two(size) && offset >= field_offset &&
+           offset + size <= field_offset + field_size;
+}
+
+bool is_valid_sock_common_access(const int offset, const int size, const AccessType access_type) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)